src/main/java/net/sf/jabref/gui/desktop/JabRefDesktop.java

```diff
@@ -309,7 +310,13 @@ public static void openConsole(File file) throws IOException {
         }
 
         String absolutePath = file.toPath().toAbsolutePath().getParent().toString();
-        NATIVE_DESKTOP.openConsole(absolutePath);
+        File console = new File(Globals.prefs.get(JabRefPreferences.CONSOLE_APPLICATION));
+
+        if (Globals.prefs.getBoolean(JabRefPreferences.USE_DEFAULT_CONSOLE_APPLICATION)) {
+            NATIVE_DESKTOP.openConsole(absolutePath);
+        } else if (console.exists() && console.isFile() && console.isAbsolute()) {
+            Runtime.getRuntime().exec(console.getAbsolutePath(), null, new File(absolutePath));
+        }
```

Comment:
Please add a java doc comment. Also mention that this depends on the user settings.